garage/envs/base.py

```diff
@@ -1,17 +1,132 @@
+"""Wrapper class that converts gym.Env into GarageEnv."""
 import collections
 
-from cached_property import cached_property
+import gym
+from gym.spaces import Box as GymBox
+from gym.spaces import Discrete as GymDiscrete
+from gym.spaces import Tuple as GymTuple
+
+from garage.core import Parameterized
+from garage.core import Serializable
+from garage.spaces import Box
+from garage.spaces import Discrete
+from garage.spaces import Product
+
+
+class GarageEnv(gym.Wrapper, Parameterized, Serializable):
+    """
+    Returns a Garage wrapper class for gym.Env.
+
+    In order to provide pickling (serialization) and parameterization
+    for gym.Envs, they must be wrapped with GarageEnv. This ensures
+    compatibility with existing samplers and checkpointing when the
+    envs are passed internally around garage.
+
+    Furthermore, GarageEnv silently converts action_space and
+    observation_space from gym.Spaces to garage.spaces.
+
+    Args:
+        env (gym.Env): the env that will be wrapped
+    """
+
+    def __init__(self, env):
+        super(GarageEnv, self).__init__(env)
+        Parameterized.__init__(self)
+        Serializable.quick_init(self, locals())
+        self._wrapped_env = env
```

Comment:
All of these are already implemented by gym.Wrapper

https://github.com/openai/gym/blob/master/gym/core.py#L238